maxim/logger/groq/utils.py

```
@@ -0,0 +1,309 @@
+            if hasattr(completion, 'usage') and completion.usage:
+                parsed_response["usage"] = {
+                    "prompt_tokens": getattr(completion.usage, 'prompt_tokens', 0),
+                    "completion_tokens": getattr(completion.usage, 'completion_tokens', 0),
+                    "total_tokens": getattr(completion.usage, 'total_tokens', 0),
+                }
+
+            return parsed_response
+
+        # Fallback for dict-like responses
+        elif isinstance(completion, dict):
+            return completion
```

Ruff (0.11.9)

Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)